soh/src/code/z_en_item00.c

```diff
@@ -1363,15 +1369,6 @@ static const Vtx customDropVtx[] = {
  */
 void EnItem00_DrawCollectible(EnItem00* this, PlayState* play) {
     if (gSaveContext.n64ddFlag && (this->getItemId != GI_NONE || this->actor.params == ITEM00_SMALL_KEY)) {
-        RandomizerCheck randoCheck =
-            Randomizer_GetCheckFromActor(this->actor.id, play->sceneNum, this->ogParams);
-
-        if (randoCheck != RC_UNKNOWN_CHECK) {
-            this->randoGiEntry =
-                Randomizer_GetItemFromKnownCheck(randoCheck, GI_NONE);
-            this->randoGiEntry.getItemFrom = ITEM_FROM_FREESTANDING;
-        }
-        
```

Comment:
Why was this removed? This reverts the fix introduced by #2718
It is needed to ensure the progressive items current draw GI model is updated as other progressive items are picked up in the same scene.

Consider two beehives both drop progressive bow. If you pick one up, the other should automatically switch to the next level, or a blue rupee if maxed out (plentiful pool).

You should be able to update this to match the changes you made in the Init method.

Reply:
They were removed because the `Randomizer_GetCheckFromActor` function doesn't get the correct check and the correct one that was set thru code is erased.
I removed the entire block because I thought that it was just doing redundant work.
I have re-added the `Randomizer_GetItemFromKnownCheck` but left out the `Randomizer_GetCheckFromActor` as it's now stored in the actor instance which prevents it being overwritten for beehive items.